File: backend/backend/Areas/Main/Models/UserNotes.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Areas.Identity.Models;

namespace backend.Areas.Main.Models;

public class UserNotes
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public string UserId { get; set; }
    public User User { get; set; }
    public int NoteId { get; set; }
    public Note Note { get; set; }
}

File: frontend/lib/models/user_notes.dart
import 'package:frontend/models/User.dart';
import 'package:frontend/models/note.dart';

class UserNotes {
  int? id;
  User? user;
  Note? note;
  String? userId;
  int? noteId;

  UserNotes({
    this.id,
    this.user,
    this.note,
    this.userId,
    this.noteId
  });

  factory UserNotes.fromJson(Map<String, dynamic> json) {
    return UserNotes(
      id: json['id'],
      userId: json['userId'],
      user: json['user'] != null ? User.fromJson(json['user']) : null,
      noteId: json['noteId'],
      note: json['note'] != null ? Note.fromJson(json['note']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user': user,
      'note': note,
      'noteId': noteId,
      'userId': userId
    };
  }
}